cmd/zed/zed_conf.c

```diff
@@ -31,6 +32,12 @@
 #include "zed_log.h"
 #include "zed_strings.h"
 
+#ifdef __APPLE__
+ssize_t readv(int, const struct iovec *, int);
+ssize_t writev(int, const struct iovec *, int);
+#endif
```

Comment:
Are we missing a header here?  Why the forward references?

Reply:
This actually stems from that `zed` does not use `lib/libspl/include` - and why would it, it's written for linux, we just usurped it for our own master plan.  But things fixed in the libspl headers needs to be reproduced a bit here.
`readv()` and `writev()` is defined in `sys/uio.h` on macOS - but then we get the whole `uio_t` being without `*` in XNU, and with `*` in zfs. It was by far easier to forward reference it. Normally, it would be hidden in `lib/libspl/include/os/macos`.

Reply:
I'm still not sure I follow.  Are the prototypes slightly different on macOS and that's why this is needed?  What exactly would have gotten fixed up the libspl headers?  That said, I'm mainly just curious and I think we can live with this if it's the smallest fix for this.